src/components/TxPopup/waitTransaction.ts

```diff
@@ -0,0 +1,94 @@
+// INFO: from https://ethereum.stackexchange.com/a/80694
+/**
+ * Wait transactions to be mined.
+ *
+ * Based on https://raw.githubusercontent.com/Kaisle/await-transaction-mined/master/index.js
+ */
+
+const DEFAULT_INTERVAL = 500
+
+const DEFAULT_BLOCKS_TO_WAIT = 0
+
+interface Options {
+	interval: number
+	blocksToWait: number
+}
+
+/**
+ * Wait for one or multiple transactions to confirm.
+ *
+ * @param library A web3 library
+ * @param txnHash A transaction hash or list of those
+ * @param options Wait timers
+ * @return Transaction receipt
+ */
+export function waitTransaction(library: any, txnHash: string | string[], options: Options = null): Promise<any> {
```

Comment:
Options aren't passed in the popup, so unless there is a good reason maybe remove it alltogether?

Then it can be change to something like this

`export function waitTransaction(library: any, txnHash: string | string[]): Promise<any> {

	const transactionReceiptAsync = async function (txnHash: any, resolve: any, reject: any) {

		library.eth.getTransactionReceipt(txnHash, (error: any, receipt: any) => {
			if (error) {
				reject(error);
			} else if (receipt == null) {
				setTimeout(
					() => transactionReceiptAsync(txnHash, resolve, reject),
					DEFAULT_INTERVAL );
			} else {
				resolve(receipt)
			}
		});
	}

	// Resolve multiple transactions once
	if (Array.isArray(txnHash)) {
		const promises: any[] = []
		txnHash.forEach(function (oneTxHash) {
			promises.push(waitTransaction(library, oneTxHash))
		})
		return Promise.all(promises)
	} else {
		return new Promise(function (resolve, reject) {
			transactionReceiptAsync(txnHash, resolve, reject)
		})
	}
}`

Reply:
We may not use it right now, but it could come in handy later. There could be instances where we want to wait for a certain number of confirmations. Let's leave it in there, so the functionality is there.

Reply:
hi @Daizze thanks for the review and the note. I think we'll leave the options as they are for this bit of code because we may want to use them in the future, and this functionality will probably be moved into `baolib` at some point so that the rest of the site can use it. i'm just leaving it here in the `TxPopup` folder for now because that's where I need it for this PR.